models/commons/src/reference/transformers.ts

```diff
@@ -1,14 +1,23 @@
 import { Transformer } from '@commercetools-test-data/core';
 import omit from 'lodash/omit';
-import type { TReference, TReferenceGraphql, TReferenceRest } from './types';
+import type {
+  TExpandedReferenceObject,
+  TReference,
+  TReferenceGraphql,
+  TReferenceRest,
+} from './types';
 
 const transformers = {
   default: Transformer<TReference, TReference>('default', {}),
   rest: Transformer<TReference, TReferenceRest>('rest', {
-    replaceFields: ({ fields }) => ({
-      ...fields,
-      obj: omit(fields, ['typeId']),
-    }),
+    buildFields: ['obj'],
+    replaceFields: ({ fields }) => {
+      return {
+        ...fields,
+        obj:
+          (fields.obj as TExpandedReferenceObject) || omit(fields, ['typeId']),
+      };
+    },
```

Comment:
This is so we can forward to the `obj` object all the properties we potentially have received so, in case the model was populated with a reference object, the model instance has its properties.